packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2303,6 +2333,119 @@ export class TransactionController extends BaseController<
   private onTransactionStatusChange(transactionMeta: TransactionMeta) {
     this.hub.emit('transaction-status-update', { transactionMeta });
   }
+
+  #updateSimulationDataIfNeeded(originalTransactionMeta: TransactionMeta) {
```

Comment:
I'm worried we're diverging from the latest code which we want to avoid so it's easier to maintain and rebase in future.

Can we not keep the exact previous code we had with `updateTransactionInternal` but just do the callback update manually rather than via the base controller?